### PR TITLE
Fix mkbundle executable crash

### PR DESCRIPTION
**Pull request by memchr**
In `CKAN.GameInstance.PortableDir (CKAN.Games.IGame game)`:

```cs
            // Find the directory our executable is stored in.
            string exeDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
````

An empty string would be assigned to `exeDir` when ckan is packaged
using mkbundle. This caused following method call
`game.GameInFolder(new DirectoryInfo(exeDir))` to crash with
`System.ArgumentException: The specified path is not of a legal form (empty).`

```
mkbundle --deps -o ckan --simple --machine-config /etc/mono/4.5/machine.config --config /etc/mono/config --simple ckan.exe
```

### CODE REVIEW

File: Core/GameInstance.cs [hunk 1]
```diff
@@ -241,6 +241,10 @@ public static string PortableDir(IGame game)
 
             // Find the directory our executable is stored in.
             string exeDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+			if (exeDir.Length == 0)
+			{
+				return null;
+			}
```
Comment (HebaruSan) [inline Core/GameInstance.cs:247]:
```suggestion
            if (exeDir.Length == 0)
            {
                return null;
            }
```

Please keep the indentation consistent with the existing code.

**Comment by HebaruSan** [comment]
For anyone else who wasn't familiar with `mkbundle`, it generates "self-contained executables that do not rely on Mono being installed on the system to simplify deployment of .NET Applications":

- https://www.mono-project.com/docs/tools+libraries/tools/mkbundle/

That sounds potentially useful as an alternate download format. Though for this specific fix, it would be better if we could fall back to the path of the mkbundle-generated executable when the executing assembly path can't be found, so users could still put their native-compiled `ckan` executable in the game root.

@memchr, would you mind checking the size of the executables generated in this way? I'm not currently in a Mono environment.

**Comment by HebaruSan** [comment]
This might be the way to get the bundle path:

- https://stackoverflow.com/a/57649113/2422988

```
Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName)
```

**Comment by memchr** [comment]
> would you mind checking the size of the executables generated in this way? I'm not currently in a Mono environment.

Size of executable generated with mono `6.12.0.177` on Linux is **36.1 MB**

**Comment by HebaruSan** [comment]
Thanks! That's surprisingly reasonable compared to other modern day downloads (even if it's much larger than `ckan.exe` by itself). Bigger than Python, smaller than VLC or Git. This might make a nice middle ground between the current downloads and the full-on bloat packaging requested in #2636.

**Comment by HebaruSan** [comment]
What command line are you using to build? My Ubuntu VM is giving me an error related to `I18N.dll`, but if you've already got it building enough to crash, maybe you already know the solution to this?

```
$ mkbundle --deps ckan.exe
Failure to load i18n assemblies, the following directories were searched for the assemblies:
   Path: .
In Custom mode, you need to provide the directory to lookup assemblies from using -L
ERROR: Couldn't load one or more of the i18n assemblies: Failed to load I18N.dll
```

**Comment by memchr** [comment]
The command i am using is:
```bash
mkbundle --deps -o ckan --simple --machine-config /etc/mono/4.5/machine.config --config /etc/mono/config --simple ckan.exe
```

**Comment by HebaruSan** [comment]
Thanks, that also worked on Ubuntu. 🎉

**Review by HebaruSan — APPROVED** [review]
Thanks for improving compatibility with `mkbundle`!

**Pull request merged**